src/wp-includes/rest-api/endpoints/class-wp-rest-block-directory-controller.php

```diff
@@ -0,0 +1,335 @@
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Search and retrieve blocks metadata
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
+		require_once ABSPATH . 'wp-admin/includes/plugin-install.php';
+		require_once ABSPATH . 'wp-admin/includes/plugin.php';
+
+		$response = plugins_api(
+			'query_plugins',
+			array(
+				'block'    => $request['term'],
+				'per_page' => $request['per_page'],
+				'page'     => $request['page'],
+			)
+		);
+
+		if ( is_wp_error( $response ) ) {
+			$response->add_data( array( 'status' => 500 ) );
+
+			return $response;
+		}
+
+		$result = array();
+
+		foreach ( $response->plugins as $plugin ) {
+			if ( $this->find_plugin_for_slug( $plugin['slug'] ) ) {
+				continue;
+			}
+
+			$data     = $this->prepare_item_for_response( $plugin, $request );
+			$result[] = $this->prepare_response_for_collection( $data );
+		}
+
+		return rest_ensure_response( $result );
+	}
+
+	/**
+	 * Parse block metadata for a block, and prepare it for an API repsonse.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param array           $plugin  The plugin metadata.
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function prepare_item_for_response( $plugin, $request ) {
+		// There might be multiple blocks in a plugin. Only the first block is mapped.
+		$block_data = reset( $plugin['blocks'] );
+
+		// A data array containing the properties we'll return.
+		$block = array(
+			'name'                => $block_data['name'],
+			'title'               => ( $block_data['title'] ? $block_data['title'] : $plugin['name'] ),
+			'description'         => wp_trim_words( $plugin['description'], 30, '...' ),
+			'id'                  => $plugin['slug'],
+			'rating'              => $plugin['rating'] / 20,
+			'rating_count'        => intval( $plugin['num_ratings'] ),
+			'active_installs'     => intval( $plugin['active_installs'] ),
+			'author_block_rating' => $plugin['author_block_rating'] / 20,
+			'author_block_count'  => intval( $plugin['author_block_count'] ),
+			'author'              => wp_strip_all_tags( $plugin['author'] ),
+			'icon'                => ( isset( $plugin['icons']['1x'] ) ? $plugin['icons']['1x'] : 'block-default' ),
+			'assets'              => array(),
+			'last_updated'        => $plugin['last_updated'],
+			'humanized_updated'   => sprintf(
+				/* translators: %s: Human-readable time difference. */
+				__( '%s ago' ),
+				human_time_diff( strtotime( $plugin['last_updated'] ) )
+			),
+		);
+
+		foreach ( $plugin['block_assets'] as $asset ) {
+			// TODO: Return from API, not client-set.
```

Comment:
See https://github.com/WordPress/gutenberg/pull/23499

Reply:
Fixed in 88f6e77.